### PR TITLE
KCC setup

### PR DESCRIPTION
**Pull request by scottilee**
This is the YAML and scripts to setup config connector.

I'm pretty bad with IAM permissions and I am getting the following error, would appreciate help:
```
$ kubectl describe containercluster -n kf-community

containercluster-controller  Update call failed: error reading underlying resource: Error reading Container Cluster "kf-community": googleapi: Error 403: Required "container.clusters.get" permission(s) for "projects/kf-community/zones/us-west1-a/clusters/kf-community"., forbidden
```

Related to #457

/assign @jlewi

---
This change is [Reviewable](https://reviewable.io/reviews/kubeflow/testing/525)

### CODE REVIEW

File: config_connector/README.md [hunk 1]
```diff
@@ -0,0 +1,9 @@
+# Cloud Connector
+
+This directory contains configurations and scripts for using [Config Connector](https://cloud.google.com/config-connector/) to setup Kubernetes clusters.
+
+## Setup
+
+Run `config_connector_setup.sh` to setup Config Connector on a GKE cluster.
+
```
Comment (jlewi) [inline config_connector/README.md]: Can you add more information about how this is setup?

e.g. what the folder and project organization is? What information would a person need to know in order to fix any problems and continue to iterate on this?

**Comment by jlewi** [comment]
Does the project kf-community exist?

**Comment by scottilee** [comment]
@jlewi No, because of the error above it is not actually creating.

**Comment by jlewi** [comment]
@scottilee the error above indicates a problem creating the cluster which I suspect is because the project doesn't exist. 

How is the project being created? Are you using KCC to create the project?

**Comment by scottilee** [comment]
Yes, I am using KCC to create the project. The YAML should create the namespace and thus the project but it says I am missing the "container.clusters.get" permission.

**Comment by jlewi** [comment]
@scottilee I don't think KCC supports creating GCP projects right now; see GoogleCloudPlatform/k8s-config-connector#32.

When you create a namespace in the KCC cluster you are associating that namespace with the GCP project in which the KCC resources in that namespace will be created. However, I don't think that is actually creating the project.

Try creating the project kf-community manually and then setting up the IAM roles for KCC so that it has appropriate permissions to create GKE clusters in that project.

**Comment by jlewi** [comment]
@scottilee What's the current status of this PR and the associated issue #457?

**Comment by scottilee** [comment]
The kf cluster is currently on v1alpha1 of config connector (`kubectl api-versions | grep -i cnrm` shows `iam.cnrm.cloud.google.com/v1alpha1` or `container.cnrm.cloud.google.com/v1alpha2`) and using v1alpha1 versions causes the following error: `Error from server (InternalError): error when creating "iamserviceaccount.yaml": Internal error occurred: failed calling webhook "cnrm-deny-unknown-fields-webhook.google.com": Post https://cnrm-validating-webhook-service.cnrm-system.svc:443/deny-unknown-fields?timeout=30s: service "cnrm-validating-webhook-service" not found`

I'm not sure how to get config connector to use v1beta1, which is the latest version. I've tried reinstalling the latest version of config connector but it keeps showing up as v1alpha1.

**Comment by jlewi** [comment]
@scottilee Are you following the latest instructions?
https://cloud.google.com/config-connector/docs/how-to/install-upgrade-uninstall

Did you try creating a new cluster and following those updated instructions?

**Comment by scottilee** [comment]
@jlewi please review, I've got things working now.

**Comment by jlewi** [comment]
This is great @scottilee I see two clusters
* kf-community 
* kubeflow-cloudconnector

Which one of these clusters is the cluster running KCC?

Is the other one a test to verify you can create clusters using KCC?

It looks like the cluster kubeflow-cloudconnector is using a K8s secret to store the GCP service account and not workload identity?

But it looks like you are setting up workload identity? e.g. your IAM policy is setting workload identity.
https://github.com/kubeflow/testing/blob/f3863dfe040bfec9b9938ead40380c60c2cfae7b/config_connector/iampolicy.yaml

**Comment by scottilee** [comment]
@jlewi kubeflow-cloudconnector is running KCC. The other, or kf-community, is for Kubeflow Community projects (https://github.com/kubeflow/community), such as the [calendar script](https://github.com/kubeflow/community/pull/283) from a while back, and it is not a test cluster.

I don't know much about Workload Identity unfortunately, I just remember you preferring me to use when setting up Config Connector.

**Pull request merged**